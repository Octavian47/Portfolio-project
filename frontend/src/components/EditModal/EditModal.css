/* src/components/EditModal/EditModal.css */
.modal-overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5); /* Semi-transparent background */
    display: flex;
    justify-content: center; /* Horizontally centers the modal */
    align-items: center; /* Vertically centers the modal */
    z-index: 1000; /* Ensures the modal is above other content */
  }
  
  .modal-content {
    background: #fff;
    padding: 20px;
    border-radius: 8px;
    width: 500px;
    max-width: 90%;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1); /* Adds a shadow for some depth */
    position: relative; /* Ensures the modal is positioned relative to the centered overlay */
  }
  
  .modal-content form div {
    margin-bottom: 15px;
  }
  
  .modal-content label {
    display: block;
    font-weight: bold;
    margin-bottom: 5px;
  }
  
  .modal-content input,
  .modal-content textarea,
  .modal-content select {
    width: 100%;
    padding: 8px;
    box-sizing: border-box;
  }
  
  .modal-content button {
    margin-right: 10px;
    padding: 12px 20px;
    border: none;
    background-color: #2f6d45;
    color: white;
    cursor: pointer;
    border-radius: 4px;
    transition: background-color 0.3s ease;
  }
  
  .modal-content button[type="button"] {
    background-color: #6c757d;
  }
  
  .modal-content button:hover {
    background-color: #255a39;
  }
  
  .modal-content button[type="button"]:hover {
    background-color: #5a6268;
  }
  