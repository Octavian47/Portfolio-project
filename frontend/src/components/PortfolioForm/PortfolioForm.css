/* src/components/PortfolioForm/PortfolioForm.css */

.message {
    padding: 10px;
    margin: 10px 0;
    border-radius: 5px;
    font-size: 16px;
  }
  
  .message.success {
    color: green;
    background-color: #e7f9e7;
  }
  
  .message.warning {
    color: orange;
    background-color: #fff4e5;
  }
  
  .message.error {
    color: red;
    background-color: #f9e7e7;
  }
  