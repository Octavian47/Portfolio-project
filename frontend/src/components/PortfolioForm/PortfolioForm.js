import React, { useState } from 'react';

const PortfolioForm = ({ onSubmit, initialData = {} }) => {
  const [title, setTitle] = useState(initialData.title || '');
  const [description, setDescription] = useState(initialData.description || '');
  const [clientUrl, setClientUrl] = useState(initialData.clientUrl || 'https://');
  const [status, setStatus] = useState(initialData.status || 'display');
  const [imageFile, setImageFile] = useState(null);

  const handleSubmit = (e) => {
    e.preventDefault();
    const formData = {
      title,
      description,
      clientUrl,
      status,
    };
    onSubmit(formData, imageFile);
  };

  return (
    <form onSubmit={handleSubmit} className="portfolio-form">
      <div>
        <label>Title:</label>
        <input
          type="text"
          value={title}
          onChange={(e) => setTitle(e.target.value)}
          required
        />
      </div>
      <div>
        <label>Description:</label>
        <textarea
          value={description}
          onChange={(e) => setDescription(e.target.value)}
          required
        />
      </div>
      <div>
        <label>Client URL:</label>
        <input
          type="url"
          value={clientUrl}
          onChange={(e) => setClientUrl(e.target.value)}
          placeholder="https://www.example.com"
          required
        />
        <small>Please include "https://" or "http://" in the URL.</small>
      </div>
      <div>
        <label>Status:</label>
        <select
          value={status}
          onChange={(e) => setStatus(e.target.value)}
          required
        >
          <option value="display">Display</option>
          <option value="hidden">Hidden</option>
        </select>
      </div>
      <div>
        <label>Image:</label>
        <input
          type="file"
          onChange={(e) => setImageFile(e.target.files[0])}
          required
        />
      </div>
      <button type="submit">Save</button>
    </form>
  );
};

export default PortfolioForm;
