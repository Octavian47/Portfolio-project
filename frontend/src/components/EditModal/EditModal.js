import React from 'react';
import PortfolioForm from '../PortfolioForm/PortfolioForm';
import './EditModal.css';

const EditModal = ({ isOpen, onClose, item, onSave }) => {
  if (!isOpen) {
    return null;
  }

  const handleFormSubmit = (formData) => {
    onSave({ ...item, ...formData });
    onClose(); // Close the modal after saving
  };

  return (
    <div className="modal-overlay">
      <div className="modal-content">
        <h2>Edit Portfolio Item</h2>
        <PortfolioForm initialData={item} onSubmit={handleFormSubmit} />
        <button onClick={onClose}>Cancel</button>
      </div>
    </div>
  );
};

export default EditModal;
