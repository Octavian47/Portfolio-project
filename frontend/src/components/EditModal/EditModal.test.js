import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import EditModal from './EditModal';
import PortfolioForm from '../PortfolioForm/PortfolioForm';

// Mock the PortfolioForm component to isolate testing for EditModal
jest.mock('../PortfolioForm/PortfolioForm', () => ({ onSubmit }) => (
  <form onSubmit={(e) => { e.preventDefault(); onSubmit({ title: 'New Title' }); }}>
    <input type="text" name="title" defaultValue="Existing Title" />
    <button type="submit">Save</button>
  </form>
));

describe('EditModal Component', () => {

  const mockItem = {
    id: '1',
    title: 'Existing Title',
    description: 'Existing Description',
    imageFilename: '/path/to/image.jpg',
    clientUrl: 'https://www.example.com',
    status: 'visible',
  };

  it('does not render when isOpen is false', () => {
    const { queryByText } = render(
      <EditModal isOpen={false} onClose={jest.fn()} item={mockItem} onSave={jest.fn()} />
    );

    // Check that the modal content is not rendered
    expect(queryByText('Edit Portfolio Item')).not.toBeInTheDocument();
  });

  it('renders correctly when isOpen is true', () => {
    const { getByText } = render(
      <EditModal isOpen={true} onClose={jest.fn()} item={mockItem} onSave={jest.fn()} />
    );

    // Check that the modal content is rendered
    expect(getByText('Edit Portfolio Item')).toBeInTheDocument();
    expect(getByText('Cancel')).toBeInTheDocument();
  });

  it('calls onSave with the correct data when the form is submitted', () => {
    const handleSave = jest.fn();
    const handleClose = jest.fn();

    const { getByText } = render(
      <EditModal isOpen={true} onClose={handleClose} item={mockItem} onSave={handleSave} />
    );

    // Simulate form submission
    fireEvent.click(getByText('Save'));

    // Check that onSave is called with the updated item data
    expect(handleSave).toHaveBeenCalledWith({
      ...mockItem,
      title: 'New Title',
    });

    // Check that onClose is called after saving
    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('calls onClose when the Cancel button is clicked', () => {
    const handleClose = jest.fn();

    const { getByText } = render(
      <EditModal isOpen={true} onClose={handleClose} item={mockItem} onSave={jest.fn()} />
    );

    // Simulate click on Cancel button
    fireEvent.click(getByText('Cancel'));

    // Check that onClose is called
    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <EditModal isOpen={true} onClose={jest.fn()} item={mockItem} onSave={jest.fn()} />
    );

    // Match the component against the snapshot
    expect(asFragment()).toMatchSnapshot();
  });

});
