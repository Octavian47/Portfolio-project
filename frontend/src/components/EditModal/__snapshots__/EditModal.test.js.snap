// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditModal Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-overlay"
  >
    <div
      class="modal-content"
    >
      <h2>
        Edit Portfolio Item
      </h2>
      <form>
        <input
          name="title"
          type="text"
          value="Existing Title"
        />
        <button
          type="submit"
        >
          Save
        </button>
      </form>
      <button>
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
